src/main/java/pl/put/poznan/buildingInfo/rest/BuildingInfoController.java

```diff
@@ -37,10 +40,18 @@
     @RequestMapping(value = "/new", method = RequestMethod.POST, produces = "application/json")
     public String createBuilding(@RequestBody String payload) {
         Building building = gson.fromJson(payload, Building.class);
+        // verify if location ids corresponds to assumed assumptions for key structure
         if (!Janitor.verifyBuildingStructure(building)) {
             Response res = new Response("Couldn't create a new building");
             return gson.toJson(res);
         }
+        // verify if locations are unique is unique among all buildings
```

Comment:
'Are uniqe is unique' I think that's not what you intended to write

Reply:
yup, typing faster than thinking xd